dle/search/services.py

```diff
@@ -27,40 +27,58 @@ def validate_search(request_query_params_dict: QueryDict) -> SearchRequest:
     else:
         raise InvalidSearchRequest("Search request is malformed")
 
+def build_match_query(search_query: str) -> str:
+    return """
+        AND match(ps.section_text) AGAINST (
+                %(search_query)s IN NATURAL LANGUAGE MODE
+            )    
+           """
+
 
 def process_search(search_request: SearchRequest) -> List[DrugLabel]:
     search_filter_mapping = {
         "select_section": "section_name",
         "select_agency": "source",
         "manufacturer_input": "marketer",
-        "marketing_category_input": "marketer",
         "generic_name_input": "generic_name",
         "brand_name_input": "product_name",
     }
     search_request_dict = search_request._asdict()
-    sql_params = {"search_text": search_request.search_text}
+    search_query = search_request.search_text.strip('"').lower()
     raw_sql = """
         SELECT
-            DISTINCT dl.*
+            DISTINCT dl.id,
+            dl.source,
+            dl.product_name,
+            dl.generic_name,
+            dl.version_date,
+            dl.source_product_number,
+            dl.marketer,
+            dl.link
         FROM
             data_druglabel as dl
         JOIN data_labelproduct as lp ON dl.id = lp.drug_label_id
         JOIN data_productsection as ps ON lp.id = ps.label_product_id
         WHERE
-            match(ps.section_text) AGAINST (
-                %(search_text)s IN NATURAL LANGUAGE MODE
-            )
-    """
+            1=1
+        """
+
+    # exact match query
+    if '"' in search_request.search_text:
+        raw_sql += " AND LOWER(ps.section_text) LIKE CONCAT('%%',%(search_query)s,'%%') "
+    else:
+        raw_sql += build_match_query(search_query)
+
+    sql_params = {"search_query": search_query}
     for k, v in search_request_dict.items():
         if v and k in search_filter_mapping:
             param_key = search_filter_mapping[k]
             if k == "select_section" and not v:
                 continue
             sql_params[param_key] = v
-            additional_filter = f"AND LOWER({param_key}) = LOWER(%({param_key})s) "
+            additional_filter = f"AND LOWER({param_key}) = %({param_key})s "
             raw_sql += additional_filter
     raw_sql += "LIMIT 30" #can remove this once we're done testing
```

Comment:
Is this the section limit you were worried about? Can you change it to limit 125?

Reply:
no that's just to limit the search results to the top 30. The section limit is just hardcoded at the moment in the `type_ahead_mapping`

Reply:
Ok. You can probably remove it now
```
mysql> SELECT count(DISTINCT section_name) FROM data_productsection;
+------------------------------+
| count(DISTINCT section_name) |
+------------------------------+
|                           97 |
+------------------------------+
```